### ISSUE TITLE
Cannot pass as argument

### ISSUE THREAD

**Issue by gossi** [tildeio/ember-element-helper#18]
I thought I'd be clever and pass the `{{element}}` as a primitive to my component, would look like this:

```hbs
<MyBuilder @element={{element "aside"}}>
```

but got this message instead:

```bash
Error: the `element` helper cannot be appended to the DOM directly
```

which relates to these lines:

https://github.com/tildeio/ember-element-helper/blob/67c07801ca234d4bb1463ec8d5bb7b86dcca13b6/lib/element-helper-syntax-plugin.js#L32-L36

I am now wondering if my idea is a valid use-case?

**Comment by chancancode** [comment]
@gossi That seems valid to me, so I would say it's a bug. The problem is that it didn't differentiate between `<div>{{element "p"}}</div>` vs `<MyComponent @arg={{element "p"}} />`, as both of those are considered a mustache position. We can fix this by disabling the check when entering an attr node, but in hind sight, I'm don't know why `<div>{{element "p"}}</div>` should be disallowed, assuming it is possible to make it work.

**Comment by gossi** [comment]
I wouldn't mind sending a PR for a fix of this.

Question here: Are you saying the whole check is obsolete or only under the condition of being in an attr node?

**Comment by chancancode** [comment]
Under the specific attr condition, for sure. But yeah I am pondering if the whole check can be deleted. I think you can try deleting it and see if you can make it work in the content position. If not, then you can fall back to keeping the check but disable it for the attr position. How does that sound?

**Comment by gossi** [comment]
Sounds good to me - thanks for clarification. Will jump on it later the day 👍

**Comment by chancancode** [comment]
@gossi this is now published as [v0.3.1](https://github.com/tildeio/ember-element-helper/releases/tag/v0.3.1)!
